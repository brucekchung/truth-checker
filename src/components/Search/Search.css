.Search input {
  width: 50%;
}

.Search {
  background: grey;
}