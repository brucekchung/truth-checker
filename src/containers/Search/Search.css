.Search input {
  border: none;
  height: 20px;
  color: white;
  background: #88b4cb;
  width: 60%;
}

.Search {
  padding-bottom: 28px;
  background: #131313;
}

.Search button {
  border: none;
  height: 22px;
  width: 100px;
  padding: 0px 12px 0px 12px;
  color: white;
  background: #3a5e71;
}

.Search button:hover {
  cursor: pointer;
  background: #5ba1c6;
}