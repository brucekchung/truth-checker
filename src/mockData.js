export const mock = {
  article: {
    title: 'Trump4lyfe',
    type: 'article',
    objects: [{
      author: 'Bruce',
      date: 'today',
      siteName: 'Trump4prez4eva.com',
      text: '2016 was a great year',
      pageUrl: 'www.trumplyfe.com'
    }]
  },
  watson: {
    document_tone: {
      tone_categories: [{
        tones: [
          {score: 0.1},
          {score: 0.2},
          {score: 0.3},
          {score: 0.3},
        ]
      },
      {
        tones: [{score: 0.5}]
      }
      ]
    }
  },
  author: {
    count: 100,
    results: [{citedCount: 1}, {citedCount: 3}]
  },
  site: [
    {
      RatingIcons: [{Url: "http://www.bbb.org/ratingicons/lg-A+.png", DeviceType: "web"}]
    }
  ],
  rating: {
    website: 'A+',
    author: {searchLength: 1300},
    article: {neutral: 0.4}
  }
}

export const mockApiData = {
  url: {
    author: 'Bruce',
    text: 'stuff'
  }
}

export const mockState = {
  result: {
    error: 'stuff',
    cleanArticle: 'stuff',
    rating: 'A+'
  },
  search: {
    cleanArticle: 'stuff'
  }
}