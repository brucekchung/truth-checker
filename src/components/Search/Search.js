import React, { Component } from 'react'
import { connect } from 'react-redux'
import { cleanArticle } from '../../cleaner'
import { siteRating } from '../../helper'
import { cleanArticleAction, errorAction, ratingAction } from '../../actions/actionIndex'
import { destructureUrl, bbbRating } from '../../api'
import './Search.css'
import cleanUrl from 'url-clean'

export class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      input: ''
    }
  }

  handleInput = (e) => {
    this.setState({input: e.target.value})
  }

  handleClick = async () => {
    const url = cleanUrl(this.state.input)
    const response = await destructureUrl(url)

    console.log('url: ', this.state.input)
    console.log('cleaned: ', url)

    if (response.errorCode) {
      this.props.sendError(response)

    } else {
      const cleaned = cleanArticle(response)

      this.props.sendCleanArticle(cleaned)
      this.props.sendError(null)
      this.getRating()
    }

    this.props.history.push('./result')
    this.setState({input: ''})
  }

  getRating = async () => {
    const organization = this.props.cleanArticle.siteName
    const orgData = await bbbRating(organization)
    const websiteRating = siteRating(orgData.SearchResults)

    this.props.sendRating(
      {
        website: websiteRating,
        author: null,
        article: null,
        currency: null,
      }
    )
  }

  render() {
    return (
      <div className="Search">
        <input
          value={this.state.input}
          onChange={this.handleInput}
        />
        <button
          onClick={this.handleClick}
        >CHECK
        </button>
      </div>
    )
  }
}

const mapState = (state) => ({
  cleanArticle: state.cleanArticle
})

const mapDispatch = (dispatch) => ({
  sendError: (error) => dispatch(errorAction(error)),
  sendRating: (rating) => dispatch(ratingAction(rating)),
  sendCleanArticle: (article) => dispatch(cleanArticleAction(article)),
})

export default connect(mapState, mapDispatch)(Search)