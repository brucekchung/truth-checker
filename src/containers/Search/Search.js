import React, { Component } from 'react'
import { connect } from 'react-redux'
import { cleanArticle, cleanWatsonAnalysis, cleanAuthor, cleanSiteRating } from '../../cleaner'
import { cleanArticleAction, errorAction, ratingAction } from '../../actions/actionIndex'
import { destructureUrl, bbbRating, watsonToneAnalysis, googleAuthor } from '../../api'
import './Search.css'
import { func, object } from 'prop-types'
import cleanUrl from 'url-clean'

export class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      input: ''
    }
  }

  handleInput = (event) => {
    this.setState({input: event.target.value})
  }

  handleClick = async () => {
    const url = cleanUrl(this.state.input)
    const response = await destructureUrl(url)

    // console.log('url: ', this.state.input)
    // console.log('cleaned: ', url)

    if (response.errorCode) {
      this.props.sendError(response)

    } else {
      const cleaned = cleanArticle(response)
      console.log('props: ', this.props)
      this.props.sendCleanArticle(cleaned)
      this.props.sendError(null)
      this.getRating()
    }

    this.props.history.push('./result')
    this.setState({input: ''})
  }

  getRating = async () => {
    const organization = this.props.cleanArticle.siteName
    const articleText = this.props.cleanArticle.text
    const author = this.props.cleanArticle.author
    // running all async elements at once:
    const orgData = bbbRating(organization)
    const watsonAnalysis = watsonToneAnalysis(articleText)
    const authorData = author ? googleAuthor(author) : 'none'
    // waiting to resolve all:
    const ready = await Promise.all([orgData, watsonAnalysis, authorData])

    if (ready) {
      const websiteRating = cleanSiteRating(ready[0].SearchResults)
      const articleRating = cleanWatsonAnalysis(ready[1])
      const authorRating = ready[2] === 'none' ? 'none' : cleanAuthor(ready[2])

      this.props.sendRating(
        {
          website: websiteRating,
          author: authorRating,
          article: articleRating
        }
      )
    }
  }

  render() {
    return (
      <div className="Search">
        <input
          value={this.state.input}
          onChange={this.handleInput}
        />
        <button
          onClick={this.handleClick}
        >CHECK
        </button>
      </div>
    )
  }
}

Search.propTypes = {
  cleanArticle: object,
  sendError: func,
  sendRating: func,
  sendCleanArticle: func,
  history: object
}

export const mapState = (state) => ({
  cleanArticle: state.cleanArticle
})

export const mapDispatch = (dispatch) => ({
  sendError: (error) => dispatch(errorAction(error)),
  sendRating: (rating) => dispatch(ratingAction(rating)),
  sendCleanArticle: (article) => dispatch(cleanArticleAction(article))
})

export default connect(mapState, mapDispatch)(Search)