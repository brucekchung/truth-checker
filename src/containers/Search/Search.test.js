import React from 'react'
import { shallow } from 'enzyme'
import { Search, mapState, mapDispatch } from './Search'
import { mockState } from '../../mockData'

describe('Search', () => {
  let wrapper
  
  beforeEach(() => {
    wrapper = shallow(<Search />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('mapState should return the correct object', () => {
    const actual = mapState(mockState.search)
    const expected = {
      cleanArticle: 'stuff'
    }

    expect(actual).toEqual(expected)
  })

  it('mapDispatch should call dispatch', () => {
    const mockDispatch = jest.fn()
    const call = mapDispatch(mockDispatch)

    call.sendError()

    expect(mockDispatch).toHaveBeenCalled()
  })
})