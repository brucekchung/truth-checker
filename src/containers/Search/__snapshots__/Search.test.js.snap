// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          value=""
/>,
        <button
          onClick={[Function]}
>
          CHECK
</button>,
      ],
      "className": "Search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "CHECK",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "CHECK",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            value=""
/>,
          <button
            onClick={[Function]}
>
            CHECK
</button>,
        ],
        "className": "Search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "CHECK",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "CHECK",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
